src/utils/streams.js

```diff
@@ -110,18 +113,26 @@ export class Concatenator extends Writable {
   }
 
   _destroy(err: ?Error, callback: (error?: Error) => mixed) {
+    this.log.trace('_destroy()');
     this.chunks.length = 0;
     this.contents = null;
-    callback();
+
+    // Passthrough the error information, if present.
+    // This line is needed because of the slightly inconsistent
+    // signature of callback vs err, and that we can't change.
+    err = err || undefined;
+    callback(err);
```

Comment:
This was also a mistake from my side: we're supposed to forward the error to the callback. This API is so confusing and the documentation is missing some information...